### ISSUE TITLE
Mixed Transitions animation for begin and end

### ISSUE THREAD

**Issue by joy2play** [kemondori/starling_slotmachine#1]
Hi,
Your extension is very useful one for us. you done really well. now we integrated slot game with your extension and trying to change transitions animation for beginning and ending for reel animation. How can we do it please help me. we want beginning reel animation with Transitions.LINEAR effect and ending is Transitions.EASE_IN_OUT_BACK  effect.

could you help us!

Thanks,

**Comment by matteofumagalli1275** [comment]
Hi, isn't what you need just EaseOutBack? Then you can change delay and duration as you want.
Following the sample you should change just this: rotation_handler.SetTransition(Transitions.EASE_IN_OUT_BACK, 18, 4);
to
rotation_handler.SetTransition(Transitions.EASE_OUT_BACK, 18, 4);

If you need something more customized you can create your own transition function:
http://wiki.starling-framework.org/tutorials/custom_transitions
Everything else is just math.

Is my suggestion correct or i missed something?

**Comment by joy2play** [comment]
Thank you for your reply. i checked with Transitions.EASE_OUT_BACK effect but it's not good. reel spinning is very slow in mid and begin and ending is very fast. please check with your example project and give solution for me.

**Comment by matteofumagalli1275** [comment]
Hi, check out the new repository.
I remembered that i had a sample extended transitions of what i meant and added that to the sample.
Also fixed a bug with the looping of some transitions, if now you try EASE_OUT_BACK it should be better.
If you don't like the bounce it's quite easy to change as you like, it's just this math function: http://tinyurl.com/pkhou8l
You can find that in ExtendedTransitions.as

One of these days i'll fix the sync with delayed answer too, so you should probably following this.

Please tell me if this is ok.

**Comment by matteofumagalli1275** [comment]
Any news?

**Comment by joy2play** [comment]
Sorry for late reply. it's working fine now. but i'm trying to replicate as slotmania spin effect. and i want to stop every wheel very less second after spinning. currently every wheel stop based on staring transition second. can i split it?

**Comment by matteofumagalli1275** [comment]
Yeah, that's a tricky case because that would mean set 5 different transition functions.
it's better to split them as you said. just set linear rotation and at the end push a bounce effect.
I'll put an example for this in the repository soon.

**Comment by joy2play** [comment]
Yeah thank you and i'm waiting for your example

**Comment by matteofumagalli1275** [comment]
Hi, it's way trikier then i thought.
The problem isn't splitting the transition but made them with the same behaviour even if the duration is different.
So i'm doing this way: transitions starts with the same value for each wheel, then a linear transition is inflated to keep the same speed but a different behaviour when the wheel will stop. 

It required some big changes and there is stuff to clean and test, so i created another [branch](https://github.com/kemondori/starling_slotmachine/tree/beta-starling_slotmachine). There are two new sample and set parameters is the most important thing to do if you want to change the transition function.

I added a function called ForceSameBehaviour that enables the handling of this stuff. you pass it the reference values of the transition.

Another important thing: BeginTransition now has a new parameter "speed". it lets you change manually the speed pixels/seconds of the linear transition started when the point of loop is reached (an example at SampleSlotmaniaEase).

**Comment by matteofumagalli1275** [comment]
When do you need to deliver your project?
I'm busy for some month because i'm working on another project, so if you find some problem i don't know if i can fix them in time.

I plan give a better look as soon as i can (i plan to add better behaviour with delayed answers first and add the stop button handling that skip the rotation to the bounce)

**Comment by joy2play** [comment]
Thanks, i will check with new branch and update you. currently i integrated from master branch with bounce out effect. i want to delivery my project for final output on October last week. so i will happy if you give with in 40 days. thank you again for your support.

**Comment by matteofumagalli1275** [comment]
I think the sample of the beta branch will work, if not i can do another sample with a rotation handled without transition (what we always did at my company lol, i wanted to try using transition because they seemed so flexible but then tricky stuff comes out, like always D:). it wouldn't require me too much time.

**Comment by matteofumagalli1275** [comment]
Hi, did you give a try?

**Comment by joy2play** [comment]
Sorry for delay. beta branch is working well it's mostly like slotmania effect. i will check with all scenario and come back. thanks

**Comment by matteofumagalli1275** [comment]
Got it :)

**Comment by matteofumagalli1275** [comment]
Just added stop and delayed answerd handling. I have to do a slot for work, so i'll be working on this for these weeks because i would like to try this kind of rotation handling.
So there could be some updates and bugfixs if i find something wrong.

**Comment by joy2play** [comment]
Great, i will update and check it thanks

**Comment by matteofumagalli1275** [comment]
Just commited a bugfix.
I'm closing this, if new stuff come out you can open a new issue.

TL;DR for anyone else: [check the beta branch](https://github.com/kemondori/starling_slotmachine/tree/beta-starling_slotmachine)

**Comment by syedzahidabbas** [comment]
hi can i use different icons on 2nd and 3rd columns and other columns

**Comment by matteofumagalli1275** [comment]
Not sure i got what you need.
It should be easy:
`for (var i:int = 0; i < icons_strings.length; i++ )
            {
                icons.push(assets.getTexture(icons_strings[i]));
                trace("icon " + i + " loaded");
            }`
"i" is the column. Just load different textures. Know that every icon is numbered from 0 to N.
If you add one more icon to the second wheel you should keep the new one last, so that's easy to handle.
I'm from Italy too, if you write me an email we can talk in italian.

**Comment by syedzahidabbas** [comment]
Ciao
thanks a lot so nice of you, ok i will send you email.
my question is that when the reels (cols) stop and  got answer all are stop on same position,
i want that column 1 stop on set target (0,1,2) and column 2 set target (1,2,3) like other etc..
zahid
Grazie mile

**Comment by matteofumagalli1275** [comment]
Check the SlotManiaWithStop sample.

``` actionscript
public function onAnswerPressed(evt:Event):void
        {
            for (var i:int = 0; i < wheelmatrix.wheels.length; i++)
            {
                var wheel:Wheel = wheelmatrix.wheels[i];
                var rotation_handler:WheelRotationRealistic = wheelmatrix.wheels[i].rotation_handler as WheelRotationRealistic;
                rotation_handler.setTarget(0,1,2);
                rotation_handler.gotAnswer();
            }
            button_server.enabled = false;
            button_stop.enabled = true;
        }
```

Just call setTarget giving the icons you want to show in the "i" column. You can pass to setTarget a Vector.<int> too.
You don't set the position with setTarget, but the icon itself.

**Comment by syedzahidabbas** [comment]
Very nice i got it. Thanks you So much
